src/reports.cpp

```diff
@@ -885,8 +885,9 @@ static int attack_info(const reports::context& rc, const attack_type &at, config
 		std::vector<std::string> secondary_types_png;
 		bool secondary_type_png_exist = true;
 		for(auto alt_t : alt_types){
-			secondary_types_png.push_back(std::string("icons/profiles/") + alt_t + ".png~SCALE_INTO(16,16)");
-			if(!image::locator(alt_t).file_exists() && secondary_type_png_exist) {secondary_type_png_exist=false;}
+			const std::string alt_png_str = std::string("icons/profiles/") + alt_t + ".png~SCALE_INTO(16,16)";
+			secondary_types_png.push_back(alt_png_str);
+			if(!image::locator(alt_png_str).file_exists() && secondary_type_png_exist) {secondary_type_png_exist=false;}
```

Comment:
Although it would be a bigger change, I think this would be clearer to read:

```c++
		// SCALE_INTO() is needed in case the 72x72 images/misc/missing-image.png is substituted.
		const std::string range_png = std::string("icons/profiles/") + at.range() + "_attack.png~SCALE_INTO(16,16)";
		const std::string type_png = std::string("icons/profiles/") + type + ".png~SCALE_INTO(16,16)";
		std::vector<std::string> secondary_types_png;
		for(const auto& alt_t : alt_types) {
			secondary_types_png.push_back(std::string("icons/profiles/") + alt_t + ".png~SCALE_INTO(16,16)");
		}

		// If any of the images is missing, then add a text description too.
		bool all_pngs_exist = image::locator(range_png).file_exists();
		all_pngs_exist &= image::locator(type_png).file_exists();
		for(const auto& png : secondary_types_png) {
			all_pngs_exist &= image::locator(png).file_exists();
		}
		if(!all_pngs_exist) {
			str << span_color(font::weapon_details_color) << "  " << "  "
				<< range << font::weapon_details_sep
				<< lang_type << "</span>\n";
		}
```